### PR TITLE
Add Authentication to MQTT & docker-compose file

### PR DESCRIPTION
**Pull request by Krazy998**
- Small update to add MQTT authentication to code 
- Add docker-compose file example.
- Updates to README with additional variables for username and password.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,12 +15,17 @@ docker run \
     --tty \
     --restart=always \
     --name=pyrmqtt \
+    --username=username
+    --password=password
```
Comment (ps-jay) [inline README.md]: Need `\` at the end of the lines for command continuation

File: README.md [hunk 2]
```diff
@@ -15,12 +15,17 @@ docker run \
...
     --device="$(readlink -f /dev/serial/by-id/usb-Rainforest_RFA-Z106-RA-PC_RAVEn_v2.3.21-if00-port0):/serial" \
     pyrmqtt \
         --host "${MQTT_BROKER}" \
         --device /serial
 ```
-
+## Docker-compose
+```shell
+Please see docker-compose.yaml
+```
```
Comment (ps-jay) [inline README.md]: Isn't actually code, so don't put it in a code block

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    command: --host 10.0.0.1 --username username --password password --device \serial
+    devices:
+    # Required:
+    # Change /dev/ttyUSB0 to where your raven dongle is located
+      - "/dev/ttyUSB0:/serial"
+      - "/serial"
+    image: pyrmqtt
+    container_name: pyrmqtt
+    restart: always
+    tty: true
```
Comment (ps-jay) [inline docker-compose.yaml]: Missing EOL character at the end of the line (posix standard)

File: pyrmqtt.py [hunk 4]
```diff
@@ -42,6 +48,8 @@ def main():
     client = mqtt.Client()
     connected = False
 
+    client.username_pw_set(username=args.username,password=args.password)
+
```
Comment (ps-jay) [inline pyrmqtt.py:52]: It would be better to wrap this in a conditional for backward compatibility with folks who use unauthenticated MQTT

**Review by ps-jay — CHANGES_REQUESTED** [review]
Thanks for the submission!

I'd like to see this made conditional for backward compatibility

**Comment by Krazy998** [comment]
I have made the changes as suggested - let me know if that works

Note that if you leave username and password blank the default variable is None. So effectively if you leave it blank it will not pass any credentials in.

**Review by ps-jay — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**